server/src/main/java/org/opensearch/action/admin/cluster/decommission/awareness/put/DecommissionRequest.java

```diff
@@ -58,13 +76,36 @@ public DecommissionRequest setDecommissionAttribute(DecommissionAttribute decomm
         return this;
     }
 
+    /**
+     * Sets the timeout for the request
+     *
+     * @param timeout time out for the request
+     * @return this request
+     */
+    public DecommissionRequest setDelayTimeout(TimeValue timeout) {
+        this.delayTimeout = timeout;
+        return this;
+    }
+
     /**
      * @return Returns the decommission attribute key-value
      */
     public DecommissionAttribute getDecommissionAttribute() {
         return this.decommissionAttribute;
     }
 
+    public TimeValue getDelayTimeout() {
+        return this.delayTimeout;
+    }
+
+    public void setNoDelay(boolean noDelay) {
+        this.noDelay = noDelay;
+    }
+
+    public boolean isNoDelay() {
+        return noDelay;
+    }
```

Comment:
We can simplify this by keeping only delayTimeout to zero. No delay was a flag for better API experience

Reply:
Have removed the param from `[RestDecommissionAction.java](https://github.com/opensearch-project/OpenSearch/pull/4586/files#diff-65a6fad1a8dab03e81118e6dc2e25f256a4abc3a26dd9004e45709c9db3a1992)`